rosco/controller/src/Controllers.f90

```
@@ -759,23 +767,116 @@ SUBROUTINE ActiveWakeControl(CntrPar, LocalVar, DebugVar)
+                AWC_TiltYaw(Imode) = D2R*CntrPar%AWC_amp(Imode)*sin(LocalVar%Time*2*PI*CntrPar%AWC_freq(Imode) + CntrPar%AWC_clockangle(Imode)*D2R)
                 IF (CntrPar%AWC_NumModes == 1) THEN
-                    AWC_TiltYaw(2) = PI/180*CntrPar%AWC_amp(1)*cos(LocalVar%Time*2*PI*CntrPar%AWC_freq(1) + 2*CntrPar%AWC_clockangle(1)*PI/180)
+                    AWC_TiltYaw(2) = D2R*CntrPar%AWC_amp(1)*sin(LocalVar%Time*2*PI*CntrPar%AWC_freq(1) + 2*CntrPar%AWC_clockangle(1)*D2R)
                 ENDIF
-                CALL ColemanTransformInverse(AWC_TiltYaw(1), AWC_TiltYaw(2), LocalVar%Azimuth, CntrPar%AWC_harmonic(Imode), REAL(0.0,DbKi), AWC_angle)
+                
+                ! Inverse Coleman Transformation with phase offset
+                CALL ColemanTransformInverse(AWC_TiltYaw(1), AWC_TiltYaw(2), LocalVar%Azimuth, CntrPar%AWC_harmonic(Imode), CntrPar%AWC_phaseoffset*D2R, AWC_angle)
+            END DO
 
-                DO K = 1,LocalVar%NumBl ! Loop through all blades, apply AWC_angle
-                    LocalVar%PitCom(K) = LocalVar%PitCom(K) + AWC_angle(K)
-                END DO
+            DO K = 1,LocalVar%NumBl ! Loop through all blades, apply AWC_angle
+                LocalVar%PitCom(K) = LocalVar%PitCom(K) + AWC_angle(K)
             END DO
+
+            ! DEBUG VARIABLES
+            DebugVar%axisTilt_2P = AWC_TiltYaw(1)
+            DebugVar%axisYaw_2P = AWC_TiltYaw(2)
+            CALL ColemanTransform(LocalVar%BlPitch, LocalVar%Azimuth, CntrPar%AWC_harmonic(1), AWC_TiltYaw(1), AWC_TiltYaw(2))
+
+            DebugVar%axisTilt_1P = AWC_TiltYaw(1)
             DebugVar%axisYaw_1P = AWC_TiltYaw(2)
-            DebugVar%axisTilt_2P = AWC_angle(1)
+        
+        ! Closed-loop PI / PR controller
+        ELSEIF ((CntrPar%AWC_Mode == 3) .OR. (CntrPar%AWC_Mode == 4)) THEN
+
+            !! For now, only works with AWC_NumModes=2
```

Should we throw an error or warning based on this comment?

https://github.com/jfrederik-nrel/ROSCO/blob/a0bfc4b44b40b97060897b671e6f1e24ec068cc9/rosco/controller/src/ReadSetParameters.f90#L1640

---

It's actually a bit more complicated than that. It only _sometimes_ works. If we really want to, we can still throw a warning, but it would be a lot of if statements.